gtm/src/template.js

```diff
@@ -0,0 +1,238 @@
+const log = require('logToConsole');
+const createArgumentsQueue = require('createArgumentsQueue');
+const injectScript = require('injectScript');
+const queryPermission = require('queryPermission');
+const setInWindow = require('setInWindow');
+const copyFromWindow = require('copyFromWindow');
+
+const TEMPLATE_VERSION = '1.0.0';
+const INSIGHTS_OBJECT_NAME = 'AlgoliaAnalyticsObject';
+const aa = createArgumentsQueue('aa', 'aa.queue');
+
+function isInitialized() {
+  return !!copyFromWindow(INSIGHTS_OBJECT_NAME);
+}
+
+function formatValueToList(value) {
+  // `20` is the limit that the engine processes.`
+  return value && value.split(',').slice(0, 20);
+}
+
+function logger(message, event) {
+  log('[GTM-DEBUG] Search Insights > ' + message, event || '');
+}
+
+switch (data.eventType) {
+  case 'init': {
+    if (isInitialized()) {
+      logger('The "init" event has already been called.');
+      break;
+    }
+
+    setInWindow(INSIGHTS_OBJECT_NAME, 'aa');
+
+    if (queryPermission('inject_script', data.searchInsightsEndpoint)) {
+      injectScript(
+        data.searchInsightsEndpoint,
+        data.gtmOnSuccess,
+        data.gtmOnFailure,
+        data.searchInsightsEndpoint
+      );
+    } else {
+      logger(
+        'The library endpoint is not allowed in the "Injects Scripts" permissions.\n\n' +
+          'You need to add the value: "' +
+          data.searchInsightsEndpoint +
+          '"\n\n' +
+          'See https://www.simoahava.com/analytics/custom-templates-guide-for-google-tag-manager/#step-4-modify-permissions'
+      );
+      break;
+    }
+
+    aa('addAlgoliaAgent', 'insights-gtm (' + TEMPLATE_VERSION + ')');
+
+    const initOptions = {
+      appId: data.appId,
+      apiKey: data.apiKey,
+      userHasOptedOut: data.userHasOptedOut,
+      region: data.region,
+      cookieDuration: data.cookieDuration,
+    };
+
+    logger(data.eventType, initOptions);
+
+    aa('init', initOptions);
+
+    if (data.setUserToken) {
+      aa('setUserToken', data.setUserToken);
```

Comment:
I'm not sure how this is used. It is supposed to be just text if I look at the spec of the template but here it's injected like it was a function.
Do you have a codesandbox demonstrating the usage?

Reply:
Sorry, there's a mistake here. It should be `data.userToken`. Good call!

I'm gonna edit the original post to add a live example.